server/src/main/java/org/elasticsearch/action/index/IndexRequest.java

```diff
@@ -106,7 +106,7 @@
 
     private boolean isPipelineResolved;
 
-    private boolean requireAlias;
+    private Boolean requireAlias;
```

Comment:
This should have been the case back when this was originally added. 

After backport, it will be all cool as the backport will be to 7.10.0 (where requireAlias was first introduced)

Reply:
Can't `requireAlias` be set to false for versions that don't support it rather than using a 3-value `Boolean` for it?

Reply:
If it was set to false, there would be no way to know if it was purposefully set to false or it was just the default value. 

Having the nullable value allows us to know if the caller specifically set it and thus shouldn't be overridden by the bulk request default values.